stdlib/public/Concurrency/AsyncSequence.swift

```diff
@@ -12,16 +12,112 @@
 
 import Swift
 
+/// A type that provides asynchronous, sequential, iterated access to its
+/// elements.
+///
+/// An `AsyncSequence` resembles the `Sequence` type --- offering a list of
+/// values you can step through one at a time --- and adds asynchroncity. An
+/// `AsyncSequence` may have all, some, or none of its values available when
+/// you first use it. Instead, you use `await` to receive values as they become
+/// available.
+///
+/// As with `Sequence`, you typically iterate through an `AsyncSequence` with a
+/// `for await`-`in` loop. However, because the caller must potentially wait for values,
+/// you use the `await` keyword. The following example shows how to iterate
+/// over `Counter`, a custom `AsyncSequence` that produces `Int` values from
+/// `1` up to a `howHigh` value:
+///
+///     for await i in Counter(howHigh: 10) {
+///       print(i, terminator: " ")
```

Comment:
Four space indent for doc comments, here and elsewhere in this PR.

Reply:
Corrected spacing.